crates/pallets/starknet/src/lib.rs

```diff
@@ -552,6 +563,15 @@ pub mod pallet {
             // This ensures that the function can only be called via unsigned transaction.
             ensure_none(origin)?;
 
+            // Check if class hash is whitelisted
+            let whitelisted_class_hashes = Self::whitelisted_class_hashes();
+            if !whitelisted_class_hashes.is_empty() {
```

Comment:
should we have a boolean variable for whitelist class hash enabled? i can think there would be cases where a chain starts with whitelisted but later opens it to everyone. i guess Pragma might want to do this as well right?

Reply:
Yes would make sense